test/box-luatest/lua_memory_test.lua

```diff
@@ -0,0 +1,81 @@
+local server = require('luatest.server')
+local t = require('luatest')
+local g = t.group('lua_memory setting test')
+local treegen = require('test.treegen')
+local helpers = require('test.config-luatest.helpers')
+local justrun = require('test.justrun')
+local yaml = require('yaml')
+
+local MB = 1024 * 1024
+local LUA_MIN_MEMORY = 256 * MB
+
+g.before_all(function()
+    treegen.init(g)
+end)
+
+g.after_all(function()
+    treegen.clean(g)
+end)
+
+g.test_lua_memory_setting = function(g)
+    g.server = server:new{
+        alias = 'default',
+        box_cfg = {lua_memory = LUA_MIN_MEMORY}
+    }
+    g.server:start()
+
+    g.server:exec(function()
+        local MB = 1024 * 1024
+        local LUA_MIN_MEMORY = 256 * MB
+        local LUA_EXTENDED_MEMORY = 600 * MB
+
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY)
+
+        -- Increasing the limit should be ok.
+        box.cfg{lua_memory = LUA_MIN_MEMORY + 1}
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY + 1)
+
+        local err_min = 'cannot limit lua memory with values less than 256MB'
+        -- Setting the limit less than 256MB is prohibited.
+        t.assert_error_msg_equals(err_min, box.cfg,
+            {lua_memory = LUA_MIN_MEMORY - 1})
+
+        box.cfg{lua_memory = LUA_EXTENDED_MEMORY}
+
+        local used = collectgarbage('count') * 1024
+        local free = LUA_EXTENDED_MEMORY - used
+        local a = string.rep('a', free / 3)
```

Comment:
It is better to use `_` for unused variable in Lua. See [this CI failure](https://github.com/tarantool/tarantool/actions/runs/9994276601/job/27623665785?pr=10223#step:6:999)

Reply:
Applied.